src/I18n/DateFormatTrait.php

```diff
@@ -101,7 +101,7 @@ protected function _formatObject(
             static::$formatters[$key] = $formatter;
         }
 
-        return (string)static::$formatters[$key]->format($date->format('U'));
+        return (string)static::$formatters[$key]->format($date->format('U.u'));
```

Comment:
```suggestion
        return (string)static::$formatters[$key]->format($date);
```
I ran the tests locally and directly using the `$date` instance works fine.